src/utils/embedded/embedded.provider.tsx

```diff
@@ -1275,18 +1373,106 @@ export function EmbeddedProvider({ children }: EmbeddedProviderProps) {
 
         setAuthTokenHeader(accessToken);
 
-        initEmbeddedWallet(user?.id || null, dbSession);
+        // Check for passkey authentication with our custom session
+        const needsWalletActivation =
+          localStorage.getItem("needsWalletActivation") === "true";
+        const customSessionId = localStorage.getItem("sessionId");
+        const customUserId = localStorage.getItem("userId");
+        const isCustomAuth = localStorage.getItem("isCustomAuth") === "true";
+
+        // Create a local variable for auth provider type
+        let effectiveAuthProviderType: AuthProviderType | null =
+          authProviderType;
+
+        // If we have passkey auth with custom session, set it up
+        if (
+          (needsWalletActivation || isCustomAuth) &&
+          customSessionId &&
+          customUserId &&
+          !dbSession
+        ) {
+          console.log("Detected passkey authentication with custom session");
+
+          // Build a minimal session object for wallet activation
+          dbSession = {
+            id: customSessionId,
+            userId: customUserId,
+            deviceNonce: localStorage.getItem("deviceNonce") || undefined,
+            createdAt: new Date(),
+            updatedAt: new Date(),
+            ip: "127.0.0.1",
+            userAgent: navigator.userAgent
+          };
+
+          // Clear the flag
+          localStorage.removeItem("needsWalletActivation");
+
+          // Important: Generate and set an authentication token for API requests
+          const deviceNonce =
+            localStorage.getItem("deviceNonce") || crypto.randomUUID();
+
+          // Store this nonce for future requests
+          localStorage.setItem("deviceNonce", deviceNonce);
+
+          // Create a custom auth token to use for API requests
```

Comment:
Ok, this answer the question above about the tokens not being valid. The tokens must be signed by the server, as otherwise anyone could use this custom auth header to steal someone else's session/account.